cmd/talosctl/cmd/talos/edit.go

```diff
@@ -0,0 +1,170 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package talos
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/cmd/util/editor"
+	"k8s.io/kubectl/pkg/cmd/util/editor/crlf"
+
+	"github.com/talos-systems/talos/cmd/talosctl/pkg/talos/helpers"
+	"github.com/talos-systems/talos/pkg/machinery/api/machine"
+	"github.com/talos-systems/talos/pkg/machinery/client"
+	"github.com/talos-systems/talos/pkg/resources/config"
+)
+
+var editCmdFlags struct {
+	namespace string
+	immediate bool
+}
+
+// editCmd represents the edit command.
+var editCmd = &cobra.Command{
+	Use:   "edit <type> [<id>]",
+	Short: "Edit a resource from the default editor.",
+	Args:  cobra.RangeArgs(1, 2),
+	Long: `The edit command allows you to directly edit any API resource 
+you can retrieve via the command line tools. 
+
+It will open the editor defined by your TALOS_EDITOR, 
+or EDITOR environment variables, or fall back to 'vi' for Linux 
+or 'notepad' for Windows.`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return WithClient(func(ctx context.Context, c *client.Client) error {
+			edit := editor.NewDefaultEditor([]string{
+				"TALOS_EDITOR",
+				"EDITOR",
+			})
+
+			resourceType := args[0]
+			if resourceType != config.Type {
+				return fmt.Errorf("only the config resource can be edited")
+			}
+
+			var lastError string
+
+			editFn := func(parentCtx context.Context, msg client.ResourceResponse) error {
+				if msg.Resource == nil {
+					return nil
+				}
+
+				if msg.Metadata.GetError() != "" {
```

Comment:
I believe helper function skips over items which have error set?

Reply:
yeah, right. Forgot to remove it from this place, thanks.